File: course/string/S.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.util.Locale;
import java.util.StringTokenizer;

public class S {

    private void solve() {
        readInt(); // len
        char[] hidden = readString().toCharArray();

        int m = readInt();
        char[][] words = new char[m][];
        for (int i = 0; i < m; ++i) {
            words[i] = readString().toCharArray();
        }

        int answer = getAnswer(hidden, words);
        out.println(answer);
    }

    private int getAnswer(char[] hidden, char[][] words) {
        final char unknown = '*';

        boolean[] revealed = new boolean[Character.MAX_VALUE + 1];
        for (char ch : hidden) {
            if (unknown != ch) {
                revealed[ch] = true;
            }
        }

        int[] counts = new int[Character.MAX_VALUE + 1];
        int activeWordsCount = 0;

        for (char[] word : words) {
            boolean possible = true;
            for (int i = 0; i < word.length; ++i) {
                char wordCh = word[i];
                char hiddenCh = hidden[i];

                if (unknown == hiddenCh) {
                    possible &= !revealed[wordCh];
                } else {
                    possible &= (wordCh == hiddenCh);
                }
            }

            if (!possible) continue;

            ++activeWordsCount;

            boolean[] waiting = new boolean[Character.MAX_VALUE + 1];
            for (char ch : word) {
                if (!revealed[ch]) waiting[ch] = true;
            }

            for (char ch = 'a'; ch <= 'z'; ++ch) {
                if (waiting[ch]) ++counts[ch];
            }
        }

        int ans = 0;
        for (char ch = 'a'; ch <= 'z'; ++ch) {
            if (activeWordsCount == counts[ch]) {
                ++ans;
            }
        }

        return ans;
    }

    //////////////////////////////////////////////////////////////////

    long sqrtLong(long x) {
        long root = (long)Math.sqrt(x);
        while (root * root > x) --root;
        while ((root + 1) * (root + 1) <= x) ++root;
        return root;
    }

    //////////////////////////////////////////////////////////////////

    private boolean yesNo(boolean yes) {
        return yesNo(yes, "YES", "NO");
    }

    private boolean yesNo(boolean yes, String yesString, String noString) {
        out.println(yes ? yesString : noString);
        return yes;
    }

    //////////////////////////////////////////////////////////////////

    private long readLong() {
        return Long.parseLong(readString());
    }

    private int[] readIntArray(int n) {
        int[] a = new int[n];
        for (int i = 0; i < n; ++i) a[i] = readInt();
        return a;
    }

    private int readInt() {
        return Integer.parseInt(readString());
    }

    private String readString() {
        while (!tok.hasMoreTokens()) {
            String nextLine = readLine();
            if (null == nextLine) return null;
            tok = new StringTokenizer(nextLine);
        }

        return tok.nextToken();
    }

    private String readLine() {
        try {
            return in.readLine();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    //////////////////////////////////////////////////////////////////

    private BufferedReader in;
    private StringTokenizer tok;
    private PrintWriter out;

    private void initFileIO(String inputFileName, String outputFileName) throws FileNotFoundException {
        in = new BufferedReader(new FileReader(inputFileName));
        out = new PrintWriter(outputFileName);
    }

    private void initConsoleIO() throws IOException {
        in = new BufferedReader(new InputStreamReader(System.in));
        out = new PrintWriter(System.out);
    }

    private void initIO() throws IOException {
        Locale.setDefault(Locale.US);

        String fileName = "";

        if (!fileName.isEmpty()) {
            initFileIO(fileName + ".in", fileName + ".out");
        } else {
            if (new File("input.txt").exists()) {
                initFileIO("input.txt", "output.txt");
            } else {
                initConsoleIO();
            }
        }

        tok = new StringTokenizer("");
    }

    //////////////////////////////////////////////////////////////////

    private void run() {
        try {
            long timeStart = System.currentTimeMillis();

            initIO();
            solve();
            out.close();

            long timeEnd = System.currentTimeMillis();
            System.err.println("Time(ms) = " + (timeEnd - timeStart));
        } catch (Exception e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }

    public static void main(String[] args) {
        new S().run();
    }
}

File: course/string/S.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = int64_t;
using ld = long double;
using ii = pair<int, int>;

const char SPACE = ' ', ENDL = '\n';

//////////////////////////////////////////////////////////////////

template <typename T>
T read() {
    T value;
    cin >> value;
    return value;
}

int ri() {
    return read<int>();
}

ll rll() {
    return read<ll>();
}

string rs() {
    return read<string>();
}

template<typename T>
vector<T> rv(int size) {
    vector<T> a(size);
    for (int i = 0; i < size; ++i) {
        a[i] = read<T>();
    }
    return a;
}

//////////////////////////////////////////////////////////////////

bool yn(bool result, string yes = "YES", string no = "NO") {
	cout << (result ? yes : no) << ENDL;
	return result;
}

//////////////////////////////////////////////////////////////////

vector<string> split(string& s, string seps) {
    vector<string> parts;

    string part = "";
    auto add_part = [&]() {
        parts.push_back(part);
        part = "";
    };

    for (char ch : s) {
        if (string::npos != seps.find(ch)) {
            add_part();
        } else {
            part += ch;
        }
    }

    add_part();

    return parts;
}

//////////////////////////////////////////////////////////////////

ld sqrt_ld(ld value) {
    return sqrt(value);
}

//////////////////////////////////////////////////////////////////

void init_io() {
    string file_name = "";
    if (!file_name.empty()) {
        string input_file_name = file_name + ".in";
        freopen(input_file_name.c_str(), "rt", stdin);

        string output_file_name = file_name + ".out";
        freopen(output_file_name.c_str(), "wt", stdout);
    } else {
        #ifndef ONLINE_JUDGE
        string input_file_name = "../../input.txt"; // reading from file in parent of parent of directory with file
        freopen(input_file_name.c_str(), "rt", stdin);
        #endif
    }

	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	cout << setprecision(10) << fixed;
}

int get_ans(string& hidden, vector<string>& words) {
    const char unknown = '*';

    vector<bool> revealed(1 << 8, false);
    for (char ch : hidden) {
        if (unknown != ch) {
            revealed[ch] = true;
        }
    }

    vector<int> counts(1 << 8, 0);
    int active_words_count = 0;

    for (string& word : words) {

        bool possible = true;
        for (int i = 0; i < word.length(); ++i) {
            char word_ch = word[i];
            char hidden_ch = hidden[i];
            if (unknown == hidden_ch) {
                possible &= !revealed[word_ch];
            } else {
                possible &= (word_ch == hidden_ch);
            }
        }

        if (!possible) continue;

        active_words_count++;

        vector<bool> waiting(1 << 8, false);
        for (char ch : word) {
            if (!revealed[ch]) waiting[ch] = true;
        }

        for (char ch = 'a'; ch <= 'z'; ++ch) {
            if (waiting[ch]) {
                counts[ch]++;
            }
        }
    }

    int ans = 0;

    for (char ch = 'a'; ch <= 'z'; ++ch) {
        if (active_words_count == counts[ch]) {
            ++ans;
        }
    }

    return ans;
}

void solve() {
    ri(); // len
    string hidden = rs();

    int m = ri();
    vector<string> words = rv<string>(m);

    int ans = get_ans(hidden, words);
    cout << ans << ENDL;
}

int main() {
    init_io();
    solve();
    return 0;
}